pkg/cri/resource-manager/policy/builtin/memtier/pools.go

```diff
@@ -586,28 +590,40 @@ func (p *policy) compareScores(request Request, scores map[int]Score,
 
 	// 1) a node with insufficient isolated or shared capacity loses
 	switch {
-	case isolated2 < 0 || shared2 < 0:
```

Comment:
This is the same pattern as below with the memory (see if node1 is "good", and then if node2 isn't good, node1 wins. If not, check for the opposite case). While this is right, we should maybe think how this could be done in a more readable and easy-to-validate way for all the cases we have to check.

Reply:
Hmm... I think it's not the same. The real problem with the above code snippet was that it never considered the case that both nodes fail to fullfill the preference, so it failed to handle that correctly (always blindly chose the first node). On a node without any isolated CPUs this basically screwed the whole pool sorting algorithm for all workloads eligible or annotated to prefer isolated CPUs, by practically always chosing the first node in every comparison.